scipy/spatial/distance.py

```diff
@@ -1092,26 +1156,36 @@ def sokalsneath(u, v):
 _SIMPLE_PDIST = {}
 
 for wrap_name, names, typ in [
-    ("bray_curtis", ['braycurtis'], "double"),
-    ("canberra", ['canberra'], "double"),
-    ("chebyshev", ['chebychev', 'chebyshev', 'cheby', 'cheb', 'ch'], "double"),
-    ("city_block", ["cityblock", "cblock", "cb", "c"], "double"),
-    ("euclidean", ["euclidean", "euclid", "eu", "e"], "double"),
-    ("sqeuclidean", ["sqeuclidean", "sqe", "sqeuclid"], "double"),
-    ("dice", ["dice"], "bool"),
-    ("kulsinski", ["kulsinski"], "bool"),
-    ("rogerstanimoto", ["rogerstanimoto"], "bool"),
-    ("russellrao", ["russellrao"], "bool"),
-    ("sokalmichener", ["sokalmichener"], "bool"),
-    ("sokalsneath", ["sokalsneath"], "bool"),
-    ("yule", ["yule"], "bool"),
+    ('bray_curtis', ['braycurtis'], 'double'),
+    ('canberra', ['canberra'], 'double'),
+    ('chebyshev', ['chebychev', 'chebyshev', 'cheby', 'cheb', 'ch'], 'double'),
+    ('city_block', ['cityblock', 'cblock', 'cb', 'c'], 'double'),
+    ('correlation', ['correlation', 'co'], 'double'),
+    ('cosine', ['cosine', 'cos'], 'double'),
+    ('dice', ['dice'], 'bool'),
+    ('euclidean', ['euclidean', 'euclid', 'eu', 'e'], 'double'),
+    ('kulsinski', ['kulsinski'], 'bool'),
+    ('mahalanobis', ['mahalanobis', 'mahal', 'mah'], 'double'),
+    ('minkowski',['minkowski', 'mi', 'm'], 'double'),
+    ('rogerstanimoto', ['rogerstanimoto'], 'bool'),
+    ('russellrao', ['russellrao'], 'bool'),
+    ('seuclidean', ['seuclidean', 'se', 's'], 'double'),
+    ('sokalmichener', ['sokalmichener'], 'bool'),
+    ('sokalsneath', ['sokalsneath'], 'bool'),
+    ('sqeuclidean', ['sqeuclidean', 'sqe', 'sqeuclid'], 'double'),
+    ('weighted_minkowski', ['wminkowski', 'wmi', 'wm', 'wpnorm'], 'double'),
+    ('yule', ['yule'], 'bool'),
 ]:
```

Comment:
sorted by `wrap_name` instead of `typ` and added `correlation`, `cosine`, `minkowski`, `wminkowski`, `mahalanobis`, `seuclidean` that have now default signature